app/controllers/spree/affirm_controller.rb

```diff
@@ -20,11 +20,12 @@ def confirm
         if affirm_checkout.save!
           payment = order.payments.create!({
             payment_method_id: affirm_params[:payment_method_id],
-            source: affirm_checkout
+            source: affirm_checkout,
+            amount: order.total
```

Comment:
We used to set the total amount payed based on the amount we got back from Affirm. I think that is what we want here too.

Reply:
I think we can get the total amount at affirm here by getting the charge based on the affirm token

Reply:
@peterberkenbosch I don't think so. According to https://docs.affirm.com/affirm-developers/reference#charges-authorize "Authorizing the charge generates a charge ID that will be used to reference it moving forward. You must authorize a charge to fully create it. A charge is not visible in the Read charge response, nor in the merchant dashboard until you authorize it."

I don't think it's possible to read the amount until it has been authorized. The main reason why we wouldn't want to do this as it is done here is in case someone is paying partially with one payment method and partially with another right?

Reply:
@peterberkenbosch Just wanted to follow up about what you think the solution is here. On the projects I've worked on I haven't had to deal with partial payments at all so I'm not sure exactly what needs to be handled here. Do you have any direction as far as what the requirements would be there?

Reply:
Thanks @AlistairNorman Let's follow up with some readme updates and some suggestions on how to verify the amounts.